internal/alloy/logging/slogtest.go

```diff
@@ -0,0 +1,408 @@
+// Copyright 2023 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package logging implements support for testing implementations of log/slog.Handler.
+package logging
+
+// NOTE THIS IS COPIED FROM GO AND PRIMARILY USED SINCE WE OVERRIDE THE NAME OF SOME FIELDS
```

Comment:
I would generally prefer us finding a way to not override names of fields while testing compared to copying the stdlib tests; part of the reason to use upstream tests is that we get the latest test cases whenever we update Go.

Reply:
I managed to get rid of this, the tests do a bit of hijinks but probably worth it.